functions/boss_unit.lua

```diff
@@ -35,8 +35,8 @@ local function on_entity_damaged(event)
 	entity.health = entity.health + event.final_damage_amount
 	boss.health = boss.health - event.final_damage_amount
 	if boss.health <= 0 then
+		entity.die(event.cause.force, event.cause)
 		global.boss_units[entity.unit_number] = nil
```

Comment:
```suggestion
		global.boss_units[entity.unit_number] = nil
		entity.die(event.cause.force, event.cause)
```

I get the error: 
```
LuaEntity API call when LuaEntity was invalid. in .../factorio/temp/currently-playing/functions/boss_unit.lua:31
```
when bosses die right now, You should reverse this so the entity dies after you get its unit_number.